content/lessons/chapter-5/validate-signature-1.tsx

```diff
@@ -1,40 +1,138 @@
-import { Introduction, Text } from 'ui'
+import { ScriptingChallenge, LessonInfo, CodeExample } from 'ui'
+import { EditorConfig } from 'types'
+import { useTranslations } from 'hooks'
+import { Text } from 'ui'
+import { useState } from 'react'
+import { getLessonKey } from 'lib/progress'
 
 export const metadata = {
   title: 'chapter_five.validate_signature_one.title',
-  image: '/assets/images/chapter-4-intro-1.jpg',
   key: 'CH5VLS1',
 }
+const javascript = {
+  program: `
+  console.log(encode_message(text))
+  console.log("KILL")`,
+  defaultFunction: {
+    name: 'encode_message',
+    args: ['text'],
+  },
+  defaultCode: `const { Hash } = require('crypto');
+// Provided by Vanderpoole 
+let text = "I am Vanderpoole and I have control of the private key Satoshi\\n" 
+text += "used to sign the first-ever Bitcoin transaction confirmed in block #170.\\n" 
+text += "This message is signed with the same private key." 
 
-export default function ValidateSignature1({ lang }) {
-  const t = useTranslations(lang)
-  const [objectPosition, setObjectPosition] = useState<string | undefined>()
+function encode_message(text){
+// Given an ascii-encoded text message, serialize a byte array 
+// with the Bitcoin protocol prefix string followed by the text 
+// and both components preceded by a length byte. 
+// Returns a 32-byte integer.
+const prefix = Buffer.from('Bitcoin Signed Message:\\n', 'ascii');
+
+}
+`,
+  validate: async (answer: string) => {
+    if (answer) {
+      if (answer.length === 88) {
+        return [true, 'Nicely Done ']
+      }
+      return [false, 'Not a valid 32 byte integer']
+    }
+    return [false, '']
+  },
+  constraints: [
+    {
+      range: [12, 1, 15, 1],
+      allowMultiline: true,
+    },
+  ],
+}
+
+const python = {
+  program: `
+print(encode_message(text))
+print("KILL")`,
+  defaultFunction: {
+    name: 'encode_message',
+    args: ['prefix', 'text'],
+  },
+  defaultCode: `import hashlib 
+# Defined by Bitcoin message signing protocol 
+# Provided by Vanderpoole 
+text = "I am Vanderpoole and I have control of the private key Satoshi\\n" 
+text += "used to sign the first-ever Bitcoin transaction confirmed in block #170.\\n" 
+text += "This message is signed with the same private key." 
 
-  const isMediumScreen = useMediaQuery({ width: 1024 })
+def encode_message(text): 
+  # Given an ascii-encoded text message, serialize a byte array 
+  # with the Bitcoin protocol prefix string followed by the text 
+  # and both components preceded by a length byte. 
+  # Returns a 32-byte integer.
 
-  useEffect(() => {
-    if (isMediumScreen) {
-      setObjectPosition('object-left')
-    } else {
-      setObjectPosition('object-bottom')
+`,
+  validate: async (answer: string) => {
+    if (answer) {
+      if (answer.length === 77) {
+        return [true, 'Nicely Done ']
+      }
+      return [false, 'Not a valid 32 byte integer']
     }
-  }, [isMediumScreen])
+    return [false, '']
```

Comment:
Also, the validation isn't quite right, It needs to equal exactly `52301120241471554888909249431012931756072043317860288098683158858465117082178` or else little and big endianness both are the correct answer

Reply:
alright.